fog/view/enclave/impl/src/lib.rs

```diff
@@ -296,4 +305,19 @@ where
             plaintext_shard_tx_out_search_results,
         )
     }
+
+    fn get_minimum_highest_observed_block_count(
+        &self,
+        shard_query_responses: Vec<QueryResponse>,
+    ) -> Result<u64> {
+        shard_query_responses
+            .into_iter()
+            .map(|query_response| query_response.highest_processed_block_count)
+            .min()
+            .ok_or_else(|| {
+                Error::QueryResponseCollation(
+                    "No minimum highest processed block count found.".to_string(),
+                )
+            })
```

Comment:
It looks like the only time one would get an option is when the vector is empty, perhaps something like
```suggestion
    ) -> u64 {
        shard_query_responses
            .into_iter()
            .map(|query_response| query_response.highest_processed_block_count)
            .min()
            .unwrap_or_default()
```
The reasons to suggest this:
* I think most people would understand 0 as a minimum for an empty container.
* The `QueryResponseCollation` had to be expanded for this.  It looks like it was actually unused prior
* There is a check for having more than one response in `create_client_query_response`, which means this is an error that would never propagate to the user, yet they still may have to acknowledge it.

Reply:
Done